modules/transformer-batch/src/main/scala/com/snowplowanalytics/snowplow/rdbloader/transformer/batch/ShredJob.scala

```diff
@@ -228,10 +228,11 @@ class ShredJob[T](
 
   private def sinkBad(transformed: RDD[Transformed], sinkProvider: () => BadrowSink): RDD[Transformed] =
     transformed.mapPartitionsWithIndex { case (partitionIndex, partitionData) =>
-      PartitionDataFilter.extractGoodAndSinkBad(
-        partitionData,
+      new GoodOnlyIterator(
+        partitionData.buffered,
         partitionIndex,
-        sinkProvider()
+        sinkProvider(),
+        bufferMaxSize = 10000
```

Comment:
Should we make it configurable...?

Reply:
Let's talk about the worst that can happen if it's not configurable.....

We do not expect a huge number of failed events here, so there's no need to tune buffer size for performance.  But this sink becomes important if something unexpected happens, or if the schema hits one of the weird edge cases which we know about.

The worst that can happen is the Spark Executor runs out of memory because it holds too many bad events in the in-memory buffer.  This would be a really bad thing if it happened!  Because the Spark job would never successfully complete.

There are few alternative ways to avoid that problem:

1. Make buffer size configurable, like you said.  If we find there is a memory problem then we can decrease the buffer size without making a new release.
2. Make the max size a count of _bytes_ not a count of number of events.  This avoids a problem where the default buffer size is safe for small events, but we run out of memory if the failed events are bigger than what you tested with.
3. Drop the default size of the buffer to a much smaller number.
4. More than one of the above.

Reply:
I think I'd go with 1. Looks like a nice middle ground. For now at least 😁